src/lib/support/tests/TestCHIPCounter.cpp

```diff
@@ -76,7 +76,7 @@ static int TestTeardown(void * inContext)
     return (SUCCESS);
 }
 
-int main(int argc, char * argv[])
+extern "C" int TestCHIPCounter(void)
```

Comment:
In the future, this should be handled with:

```
#ifdef __cplusplus
extern "C" {
#endif

// C or C++ Declarations

#ifdef __cplusplus
}
#endif
```

wrappers in header files rather than bespoke function decorations such as this.